devtools-har-extension/index.js

```diff
@@ -1,20 +1,59 @@
 const REQUEST_FINISHED_URL = 'chrome://version/?get-har'
+const REQUEST_RESET_URL = 'chrome://version/?reset-har'
+
+let har = {
+  'version': '1.2',
+  'creator': { 'name': '', 'version': '' },
+  'entries': [],
+  'pages': []
+}
+
+let pageRef = 0
 
 let send = (channel, msg) => {
   window.chrome.devtools.inspectedWindow.eval(
     `__nightmare.ipc.send("${channel}", ${JSON.stringify(msg)})`
   )
 }
 
-let sendHAR = () => {
-  window.chrome.devtools.network.getHAR((har) => {
-    har.entries = har.entries.filter((entry) => entry.request.url !== REQUEST_FINISHED_URL)
-    send('har-data', har)
-  })
-}
+window.chrome.devtools.network.getHAR((info) => {
+  har.creator.name = info.creator.name
+  har.creator.version = info.creator.version
+})
 
 window.chrome.devtools.network.onRequestFinished.addListener((request) => {
-  if (request.request.url === REQUEST_FINISHED_URL) {
-    sendHAR()
+  switch (request.request.url) {
+    case REQUEST_FINISHED_URL:
+      send('har-data', har)
+      break
+    case REQUEST_RESET_URL:
+      har.entries = []
+      har.pages = []
```

Comment:
I tried to use this syntax but seems that either Electron or DevTools does not support it, the code is not executed on node.